config/script.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2023 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package config
+
+import (
+	hhcl "github.com/hashicorp/hcl/v2"
+	"github.com/terramate-io/terramate/errors"
+	"github.com/terramate-io/terramate/hcl"
+	"github.com/terramate-io/terramate/hcl/eval"
+	"github.com/zclconf/go-cty/cty"
+)
+
+// Errors for indicating invalid script schema
+const (
+	ErrScriptSchema              errors.Kind = "script config has an invalid schema"
+	ErrScriptInvalidTypeDesc     errors.Kind = "invalid type for script.description"
+	ErrScriptInvalidTypeCommand  errors.Kind = "invalid type for script.command"
+	ErrScriptInvalidTypeCommands errors.Kind = "invalid type for script.commands"
+)
+
+// EvaluatedScriptJob represents an evaluated job block
+type EvaluatedScriptJob struct {
+	Command  []string
+	Commands [][]string
+}
+
+// EvaluatedScript represents an evaluated script block
+type EvaluatedScript struct {
+	Labels      []string
+	Description string
+	Jobs        []EvaluatedScriptJob
+}
+
+// CommandOrCommands is a convenience method for callers who don't specifically
+// care about which command attr was set e.g. command or commands. This method
+// returns a list of commands irrespective of whether they were set through
+// job.command or job.commands
+func (es EvaluatedScriptJob) CommandOrCommands() [][]string {
```

Comment:
Just call it `Commands()`. The evaluated data structure does not need to reflect the source code data structures. In practice, only the list of commands make sense.